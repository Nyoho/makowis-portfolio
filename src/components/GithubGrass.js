// @flow
import React from 'react';
import type { Person as Model } from '../models/Person';

export const GithubGrass = ({ github }: Model) => (
  <section className="text-center text-white bg-primary p-5">
    <h2>GitHub Grass Graph</h2>
    <img
      src={`https://grass-graph.moshimo.works/images/${github}.png`}
      alt="GitHub Grass Graph"
      className="img-fluid"
    />
    <p>
      <a href="https://grass-graph.moshimo.works">
        <span className="text-white">image from Grass-Graph</span>
      </a>
    </p>
  </section>
);

export default GithubGrass;
