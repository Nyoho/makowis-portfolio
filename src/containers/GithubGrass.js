// @flow
import { connect } from 'react-redux';

import Component from '../components/GithubGrass';
import type { State } from '../reducers/index';

const mapStateToProps = ({ person }: State) => person;

const GithubGrass = connect(mapStateToProps)(Component);

export default GithubGrass;