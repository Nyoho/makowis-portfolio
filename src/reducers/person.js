// @flow
import person from '../models/Person';
import type { Person } from '../models/Person';

const CHANGE_PERSON: 'CHANGE_PERSON' = 'CHANGE_PERSON';

export type State = Person;

type Action = { type: typeof CHANGE_PERSON, person: Person };

const reducer = (state: Person = person, action: Action): State => {
  switch (action.type) {
    case CHANGE_PERSON:
      return action.person;
    default:
      return state;
  }
};

export default reducer;
