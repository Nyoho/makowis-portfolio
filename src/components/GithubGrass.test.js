/* eslint-env jest */
// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import GithubGrass from './GithubGrass';
import person from '../models/Person';

it('renders Developers', () => {
  const tree = renderer.create(<GithubGrass {...person} />).toJSON();
  expect(tree).toMatchSnapshot();
});
