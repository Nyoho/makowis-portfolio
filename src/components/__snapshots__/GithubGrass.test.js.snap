// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Developers 1`] = `
<section
  className="text-center text-white bg-primary p-5"
>
  <h2>
    GitHub Grass Graph
  </h2>
  <img
    alt="GitHub Grass Graph"
    className="img-fluid"
    src="https://grass-graph.moshimo.works/images/makowis.png"
  />
  <p>
    <a
      href="https://grass-graph.moshimo.works"
    >
      <span
        className="text-white"
      >
        image from Grass-Graph
      </span>
    </a>
  </p>
</section>
`;
